src/pathfinder/npf/npf.cpp

```diff
@@ -1040,6 +1041,10 @@ static NPFFoundTargetData NPFRouteInternal(AyStarNode *start1, bool ignore_start
 	/* Initialize user_data */
 	_npf_aystar.user_data = user;
 
+	/* We will limit the number of nodes for now, until we have a better
+	 * solution to really fix performance */
+	AutoRestoreBackup max_search_nodes(_npf_aystar.max_search_nodes, _settings_game.pf.npf.npf_max_search_nodes);
```

Comment:
What is the purpose of restoring the value? Where/when is the restored value used?

Reply:
I'm unsure if on new game the river generator sets a new value from fresh, or re-uses the existing one. Just wanted to be safe than sorry.

Reply:
That comment makes the reasoning of this PR even weirder.

You don't know whether the river generator sets the value itself, so you remove the only thing that initialises it before the river generator runs.
The PR is about changing max search nodes not having an effect, well if river generation would be using it, it would now be using a max search nodes that was never initialised.

Even then, `_npf_aystar` is `static` in this file, so it should not be "leaking" to any other place unless that actually uses NPF. The river generator uses only `AyStar` and has its own instance allocated on the stack with infinite search nodes. So the river generator does not use the NPF one.

Reply:
`AutoRestoreBackup` is not needed for River generation, just tested 🙂.
`AyStar` has a 10000 `max search nodes` when it `Init` by default.
`NPF` uses `AyStar`, so inits with 10k.
RiverGen uses `AyStar`, so inits with 10k.
Each `NPFRouteInternal` call still sets the value of `max search nodes` to what the user has in the settings.
I guess that's safe enough for between game starts, load saves uses the saved setting.